storage/innobase/include/dict0mem.h

```diff
@@ -2256,6 +2256,11 @@ struct dict_table_t {
 	DICT_STATS_AUTO_RECALC_OFF. */
 	ib_uint32_t				stats_auto_recalc;
 
+	/** The percentage of pages to sample for this table during persistent
+	stats estimation. If this is 0, then the value of the stats_sample_pages
+        will be used instead. */
+	ulint					stats_percentage;
```

Comment:
Do we really need 32 or 64 bits for this?

The `sizeof(dict_index_t)` is already large enough. Could we combine all fields into a single word:
```c++
unsigned stat_persistent:2;
unsigned stats_auto_recalc:2;
unsigned stats_percentage:7;
```

Reply:
Where will `ha_table_option_struct::stats_percentage` be assigned to `dict_table_t::stats_percentage`? I did not find any code to do that. For the pre-existing statistics attributes that is being done in `innobase_copy_frm_flags_from_table_share()`.

Reply:
Regarding combining all fields into a single word, I tried it, but received compiler warning because of type inconsistency with definitions inside `dict0stats.inl`. I ignored the warning, but that produced failures for test case `innodb_stats_create_on_corrupted`. In the end, I opted to use `ib_uint32_t`, which made all tests pass.

>Where will ha_table_option_struct::stats_percentage be assigned to dict_table_t::stats_percentage?

Addressed in a new push.